vehicle/mkv/software/throttle/throttle.c

```diff
@@ -0,0 +1,90 @@
+#include "libs/adc/api.h"
+#include "libs/can/api.h"
+#include "libs/timer/api.h"
+#include <avr/interrupt.h>
+
+#include "throttle_config.h"
+#include "libs/gpio/api.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/*
+ * Interrupts
+ */
+ volatile bool send_can = false;
+
+ void timer0_isr(void) {
+    send_can = true;
+}
+
+void pcint0_isr(void){
+	/*
+	   Check if shutdown Circuit nodes are open or closed. 
+	 */
+	// Check value of SS_IS, SS_ESTOP, and SS_BOTS pins
+
+}
+
+int main(void) {
+	// TODO: initalize motor controller by sending CAN msg
+    can_init(BAUD_500KBPS);
+    adc_init();
+    timer_init(&timer0_cfg);
+
+    gpio_set_mode(DEBUG_LED1, OUTPUT);
+    gpio_set_mode(DEBUG_LED2, OUTPUT);
+    gpio_set_mode(DEBUG_LED3, OUTPUT);
+    gpio_set_mode(LED1, OUTPUT);
+    gpio_set_mode(LED2, OUTPUT);
+    
+
+    gpio_set_mode(SS_ESTOP, INPUT);
+    gpio_set_mode(SS_IS, INPUT);
+    gpio_set_mode(SS_BOTS, INPUT);
+
+    gpio_enable_interrupt(SS_ESTOP);
+    gpio_enable_interrupt(SS_IS);
+    gpio_enable_interrupt(SS_BOTS);
+
+	// Initialize interrupts
+	sei();
+
+	// Wait for start up sequence
+	// 	completed precharge, brakes engaged, start button pressed
+	// TODO: call receive functions for the above
+
+    while (1) {
+		if (run_1ms) {
+    	// Do 1 millisecond loop stuff
+
+		/* TODO
+		 *
+		 * - Read ADCS
+		 * 		- read the two throttle potentiometers
+		 * - Map
+		 *		- map the values from the pots to 0-255
+		 *		- if the voltage is out of range, set to min or max
```

Comment:
If it's out of range, we might want to throw a fault. Or maybe if it's out of range for too long... Lemme think on it